diff --git a/node_modules/@dcloudio/uni-mp-weixin/dist/uni.api.esm.js b/node_modules/@dcloudio/uni-mp-weixin/dist/uni.api.esm.js
index 0bb3276..a72dc27 100644
--- a/node_modules/@dcloudio/uni-mp-weixin/dist/uni.api.esm.js
+++ b/node_modules/@dcloudio/uni-mp-weixin/dist/uni.api.esm.js
@@ -1182,6 +1182,7 @@ function initGetProvider(providers) {
 }
 
 const objectKeys = [
+    'qy',
     'env',
     'error',
     'version',
@@ -1191,18 +1192,21 @@ const objectKeys = [
     'router',
     'worklet',
 ];
+function isWxKey(key) {
+    return objectKeys.indexOf(key) > -1 || typeof wx[key] === 'function';
+}
 function initWx() {
-    const WxProxyHandlers = {
-        get(target, key) {
-            if (hasOwn(target, key)) {
-                return target[key];
-            }
-            if (objectKeys.indexOf(key) > -1 || isFunction(wx[key])) {
-                return wx[key];
-            }
-        },
-    };
-    return new Proxy({}, WxProxyHandlers);
+    const newWx = {};
+    for (const key in wx) {
+        if (isWxKey(key)) {
+            // TODO wrapper function
+            newWx[key] = wx[key];
+        }
+    }
+    if (typeof globalThis !== 'undefined') {
+        globalThis.wx = newWx;
+    }
+    return newWx;
 }
 
 const mocks = ['__route__', '__wxExparserNodeId__', '__wxWebviewId__'];
@@ -1234,9 +1238,11 @@ function createSelectorQuery() {
     return query;
 }
 const wx$2 = initWx();
-const host = wx$2.getAppBaseInfo
-    ? wx$2.getAppBaseInfo().host
-    : wx$2.getSystemInfoSync().host;
+let baseInfo = wx$2.getAppBaseInfo && wx$2.getAppBaseInfo();
+if (!baseInfo) {
+    baseInfo = wx$2.getSystemInfoSync();
+}
+const host = baseInfo ? baseInfo.host : null;
 const shareVideoMessage = host && host.env === 'SAAASDK'
     ? wx$2.miniapp.shareVideoMessage
     : wx$2.shareVideoMessage;
