diff --git a/node_modules/@dcloudio/uni-mp-weixin/dist/uni.api.esm.js b/node_modules/@dcloudio/uni-mp-weixin/dist/uni.api.esm.js
index 0bb3276..d6f0ee2 100644
--- a/node_modules/@dcloudio/uni-mp-weixin/dist/uni.api.esm.js
+++ b/node_modules/@dcloudio/uni-mp-weixin/dist/uni.api.esm.js
@@ -1182,6 +1182,7 @@ function initGetProvider(providers) {
 }
 
 const objectKeys = [
+    'qy',
     'env',
     'error',
     'version',
@@ -1234,9 +1235,11 @@ function createSelectorQuery() {
     return query;
 }
 const wx$2 = initWx();
-const host = wx$2.getAppBaseInfo
-    ? wx$2.getAppBaseInfo().host
-    : wx$2.getSystemInfoSync().host;
+let baseInfo = wx$2.getAppBaseInfo && wx$2.getAppBaseInfo();
+if (!baseInfo) {
+    baseInfo = wx$2.getSystemInfoSync();
+}
+const host = baseInfo ? baseInfo.host : null;
 const shareVideoMessage = host && host.env === 'SAAASDK'
     ? wx$2.miniapp.shareVideoMessage
     : wx$2.shareVideoMessage;
