diff --git a/node_modules/@dcloudio/uni-h5/dist/uni-h5.cjs.js b/node_modules/@dcloudio/uni-h5/dist/uni-h5.cjs.js
index 3683391..e456495 100644
--- a/node_modules/@dcloudio/uni-h5/dist/uni-h5.cjs.js
+++ b/node_modules/@dcloudio/uni-h5/dist/uni-h5.cjs.js
@@ -5783,7 +5783,6 @@ const index$m = /* @__PURE__ */ defineBuiltInComponent({
       props2.scrollY ? style += "overflow-y:auto;" : style += "overflow-y:hidden;";
       return style;
     });
-    const slots_default = slots.default && slots.default();
     return () => {
       const {
         refresherEnabled,
@@ -5846,7 +5845,7 @@ const index$m = /* @__PURE__ */ defineBuiltInComponent({
         "fill": "none",
         "style": "color: #2bd009",
         "stroke-width": "3"
-      }, null)]) : null])]) : null, refresherDefaultStyle == "none" ? slots.refresher && slots.refresher() : null], 4) : null, slots_default], 512)], 4)], 512)], 512);
+      }, null)]) : null])]) : null, refresherDefaultStyle == "none" ? slots.refresher && slots.refresher() : null], 4) : null, slots.default && slots.default()], 512)], 4)], 512)], 512);
     };
   }
 });
@@ -5986,6 +5985,8 @@ function useScrollViewLoader(props2, state, scrollTopNumber, scrollLeftNumber, t
     content.value.removeEventListener("webkitTransitionEnd", __transitionEnd);
   }
   function _setRefreshState(_state) {
+    if (!props2.refresherEnabled)
+      return;
     switch (_state) {
       case "refreshing":
         state.refresherHeight = props2.refresherThreshold;
diff --git a/node_modules/@dcloudio/uni-h5/dist/uni-h5.es.js b/node_modules/@dcloudio/uni-h5/dist/uni-h5.es.js
index 365e7a3..65729b0 100644
--- a/node_modules/@dcloudio/uni-h5/dist/uni-h5.es.js
+++ b/node_modules/@dcloudio/uni-h5/dist/uni-h5.es.js
@@ -12987,7 +12987,6 @@ const ScrollView = /* @__PURE__ */ defineBuiltInComponent({
       props2.scrollY ? style += "overflow-y:auto;" : style += "overflow-y:hidden;";
       return style;
     });
-    const slots_default = slots.default && slots.default();
     return () => {
       const {
         refresherEnabled,
@@ -13050,7 +13049,7 @@ const ScrollView = /* @__PURE__ */ defineBuiltInComponent({
         "fill": "none",
         "style": "color: #2bd009",
         "stroke-width": "3"
-      }, null)]) : null])]) : null, refresherDefaultStyle == "none" ? slots.refresher && slots.refresher() : null], 4) : null, slots_default], 512)], 4)], 512)], 512);
+      }, null)]) : null])]) : null, refresherDefaultStyle == "none" ? slots.refresher && slots.refresher() : null], 4) : null, slots.default && slots.default()], 512)], 4)], 512)], 512);
     };
   }
 });
@@ -13232,6 +13231,8 @@ function useScrollViewLoader(props2, state2, scrollTopNumber, scrollLeftNumber,
     content.value.removeEventListener("webkitTransitionEnd", __transitionEnd);
   }
   function _setRefreshState(_state) {
+    if (!props2.refresherEnabled)
+      return;
     switch (_state) {
       case "refreshing":
         state2.refresherHeight = props2.refresherThreshold;
