diff --git a/node_modules/@dcloudio/webpack-uni-pages-loader/lib/platforms/mp.js b/node_modules/@dcloudio/webpack-uni-pages-loader/lib/platforms/mp.js
index d9ac7b8..ef66ad9 100644
--- a/node_modules/@dcloudio/webpack-uni-pages-loader/lib/platforms/mp.js
+++ b/node_modules/@dcloudio/webpack-uni-pages-loader/lib/platforms/mp.js
@@ -265,14 +265,10 @@ module.exports = function (pagesJson, manifestJson, project = {}) {
 
   const projectName = getPlatformProject()
 
-  const rootProjectPath =
+  const projectPath =
     projectName &&
     path.resolve(process.env.VUE_CLI_CONTEXT || process.cwd(), projectName)
 
-  const srcProjectPath = projectName && path.resolve(process.env.UNI_INPUT_DIR, projectName)
-
-  const projectPath = rootProjectPath && fs.existsSync(rootProjectPath) ? rootProjectPath : srcProjectPath
-
   if (projectPath && fs.existsSync(projectPath)) {
     // 自定义 project.config.json
     const platform = process.env.UNI_PLATFORM
@@ -293,6 +289,18 @@ module.exports = function (pagesJson, manifestJson, project = {}) {
       })
     }
 
+    if (
+      process.env.UNI_PLATFORM === 'mp-weixin' ||
+      process.env.UNI_PLATFORM === 'mp-qq'
+    ) {
+      // 微信不需要生成，其他平台做拷贝
+      return {
+        app: {
+          name: 'app',
+          content: trimMPJson(app)
+        }
+      }
+    }
     return {
       app: {
         name: 'app',
