diff --git a/node_modules/@dcloudio/uni-mp-kuaishou/dist/index.js b/node_modules/@dcloudio/uni-mp-kuaishou/dist/index.js
index 991c985..60bf47c 100644
--- a/node_modules/@dcloudio/uni-mp-kuaishou/dist/index.js
+++ b/node_modules/@dcloudio/uni-mp-kuaishou/dist/index.js
@@ -2386,7 +2386,9 @@ function parseBasePage (vuePageOptions, {
     this.$vm.$mp.query = query; // 兼容 mpvue
     this.$vm.__call_hook('onLoad', query);
   };
-  initUnknownHooks(pageOptions.methods, vuePageOptions, ['onReady']);
+  {
+    initUnknownHooks(pageOptions.methods, vuePageOptions, ['onReady']);
+  }
 
   return pageOptions
 }
